src/wdl_aid/wdl_aid.py

```diff
@@ -244,11 +244,67 @@ def gather_inputs(workflow: WDL.Workflow
     return inputs, required_inputs
 
 
+def gather_entries(name_and_bindings: List[Tuple[str, WDL.Env.Binding]],
+                   parameter_meta: Dict[str, Any], category_key: str,
+                   fallback_category: str, description_key: str,
+                   fallback_description: str,
+                   fallback_description_to_object: bool,
+                   excluded_names: List[str], required_names: List[str] = []
+                   ) -> Tuple[Dict[str, List[Dict[str, Any]]], List[str]]:
+    """
+    :param name_and_bindings: A tuple in which the first element is the
+    fully qualified name of the binding in the second element.
+    :param parameter_meta: A dictionary containing the parameter_meta
+    information.
+    :param category_key: The key used in parameter_meta for categories.
+    :param fallback_category: The default category.
+    :param description_key: The key used in parameter_meta for
+    descriptions.
+    :param fallback_description: The default description.
+    :param fallback_description_to_object: Whether or not the entire
+    object should be returned for a given object if the description
+    key is not found.
+    :param excluded_names: The names to exclude.
+    :param required_names: The names to separate into a "required" category.
+    :return: The entries and a list of names for which parameter_meta is
+    missing.
+    """
+    entries = {}
```

Comment:
The template determines if it should render a section for a given category based on whether this category is defined in the dictionary. Since defaultdict effectively makes all keys/categories defined, it will end up rendering a bunch of headers with empty sections underneath. This could be fixed by checking the length of the category instead of if it is defined, but it feels more intuitive to check for existence than length. It also looks a bit cleaner in the template imo:
`if inputs.other is defined` as opposed to `if inputs.other|length != 0`.